src/sentry/static/sentry/app/components/issueDiff/index.tsx

```diff
@@ -0,0 +1,185 @@
+    this.fetchData();
+  }
+
+  fetchData() {
+    const {baseIssueId, targetIssueId, baseEventId, targetEventId} = this.props;
+
+    // Fetch component and event data
+    Promise.all([
+      import(/* webpackChunkName: "splitDiff" */ '../splitDiff'),
+      this.fetchEventData(baseIssueId, baseEventId ?? 'latest'),
+      this.fetchEventData(targetIssueId, targetEventId ?? 'latest'),
+    ])
+      .then(([{default: SplitDiffAsync}, baseEvent, targetEvent]) => {
+        this.setState({
+          SplitDiffAsync,
+          baseEvent,
+          targetEvent,
+          loading: false,
+        });
+      })
+      .catch(() => {
+        addErrorMessage(t('Error loading events'));
+      });
+  }
+
+  toggleDiffMode = () => {
+    this.setState(
+      state => ({groupingDiff: !state.groupingDiff, loading: true}),
+      this.fetchData
+    );
+  };
+
+  fetchEventData = async (issueId: string, eventId: string) => {
+    const {orgId, project, api} = this.props;
+    const {groupingDiff} = this.state;
+
+    let paramEventId = eventId;
+
+    if (eventId === 'latest') {
+      const event = await api.requestPromise(`/issues/${issueId}/events/latest/`);
+      paramEventId = event.eventID;
+    }
+
+    if (groupingDiff) {
+      const groupingInfo = await api.requestPromise(
+        `/projects/${orgId}/${project.slug}/events/${paramEventId}/grouping-info/`
+      );
+      return renderGroupingInfo(groupingInfo);
+    }
+
+    const event = await api.requestPromise(
+      `/projects/${orgId}/${project.slug}/events/${paramEventId}/`
+    );
+    return getStacktraceBody(event);
+  };
+
+  render() {
+    const {className, project} = this.props;
+    const {
+      SplitDiffAsync: DiffComponent,
+      loading,
+      groupingDiff,
+      baseEvent,
+      targetEvent,
+    } = this.state;
+
+    const showDiffToggle = project.features.includes('similarity-view-v2');
+
+    return (
+      <StyledIssueDiff className={className} loading={loading}>
+        {loading && <LoadingIndicator />}
+        {!loading && showDiffToggle && (
+          <HeaderWrapper>
+            <ButtonBar merged active={groupingDiff ? 'grouping' : 'event'}>
+              <Button barId="event" size="small" onClick={this.toggleDiffMode}>
+                {t('Diff stacktrace and message')}
+              </Button>
+              <Button barId="grouping" size="small" onClick={this.toggleDiffMode}>
+                {t('Diff grouping information')}
+              </Button>
+            </ButtonBar>
+          </HeaderWrapper>
+        )}
+        {!loading &&
+          DiffComponent &&
+          baseEvent.map((value, i) => (
+            <DiffComponent
+              key={i}
+              base={value}
+              target={targetEvent[i] ?? ''}
+              type="words"
+            />
+          ))}
+      </StyledIssueDiff>
+    );
+  }
+}
+
+export default withApi(IssueDiff);
+
+// required for tests which do not provide API as context
+export {IssueDiff};
```

Comment:
you could use `api={new MockApiClient()}` in the tests 😉

Reply:
I'm using mockapiclient already, I think the issue is that I cannot pass the api client if it is wrapped in `withApi`.